### DISCUSSION TITLE
How to implement a circular trajectory?

### DISCUSSION THREAD

**Discussion by Carmol1** [USC-ACTLab/crazyswarm#658] [Q&A] [answered]
Hello.

I have used the function ''goto'' and the crazyflie effectively arrives to the desired point, but if i need to make use of sin(the), cos(the) and its derivative to generate the circular trajectory, what do I need to change in the code?.

I'd appreciate your help.

**Comment by jpreiss** [comment, marked as answer]
See https://github.com/USC-ACTLab/crazyswarm/blob/master/ros_ws/src/crazyswarm/scripts/cmdVelocityCircle.py for an example. You could also use `cmdPosition` or `cmdFullState`.

That script doesn't have a `land()` command, so you should probably add one before running it!

**Reply by Carmol1** [reply]
Hello.
Thank you so much for your help, for the moment, I am used cmdFullState.
